homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,405 @@
+            vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
+            vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
+            vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(
+                ROUTE_MODE
+            ),
+            vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+            vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+        }
+    ),
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: Dict[str, Union[str, bool]],
+    async_add_entities: Callable,
+    discovery_info: None = None,
+) -> None:
+    """Set up the HERE travel time platform."""
+
+    app_id = config[CONF_APP_ID]
+    app_code = config[CONF_APP_CODE]
+    here_client = herepy.RoutingApi(app_id, app_code)
+
+    if not _are_valid_client_credentials(here_client):
+        _LOGGER.error(
+            "Invalid credentials. This error is returned if the specified token was invalid or no contract could be found for this token."
+        )
+        return
+
+    if config.get(CONF_ORIGIN_LATITUDE) is not None:
+        origin = f"{config[CONF_ORIGIN_LATITUDE]},{config[CONF_ORIGIN_LONGITUDE]}"
+    else:
+        origin = config[CONF_ORIGIN_ENTITY_ID]
+
+    if config.get(CONF_DESTINATION_LATITUDE) is not None:
+        destination = (
+            f"{config[CONF_DESTINATION_LATITUDE]},{config[CONF_DESTINATION_LONGITUDE]}"
+        )
+    else:
+        destination = config[CONF_DESTINATION_ENTITY_ID]
+
+    travel_mode = config[CONF_MODE]
+    traffic_mode = config[CONF_TRAFFIC_MODE]
+    route_mode = config[CONF_ROUTE_MODE]
+    name = config[CONF_NAME]
+    units = config.get(CONF_UNIT_SYSTEM, hass.config.units.name)
+
+    here_data = HERETravelTimeData(
+        here_client, travel_mode, traffic_mode, route_mode, units
+    )
+
+    sensor = HERETravelTimeSensor(name, origin, destination, here_data)
+
+    async_add_entities([sensor], True)
+
+
+def _are_valid_client_credentials(here_client: herepy.RoutingApi) -> bool:
+    """Check if the provided credentials are correct using defaults."""
+    test_origin = [38.9, -77.04833]
+    test_destination = [39.0, -77.1]
+    try:
+        here_client.car_route(
+            test_origin,
+            test_destination,
+            [
+                herepy.RouteMode[ROUTE_MODE_FASTEST],
+                herepy.RouteMode[TRAVEL_MODE_CAR],
+                herepy.RouteMode[TRAFFIC_MODE_DISABLED],
+            ],
+        )
+    except herepy.InvalidCredentialsError:
+        return False
+    return True
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(
+        self, name: str, origin: str, destination: str, here_data: "HERETravelTimeData"
```

Comment:
Does this work?
```suggestion
        self, name: str, origin: str, destination: str, here_data: HERETravelTimeData
```

Reply:
If I remove the quotation marks I get a flake8 error:
`F821 undefined name 'HERETravelTimeData`

Reply:
Ok. This the problem of forward forward references.

https://www.python.org/dev/peps/pep-0563/

In Python 3.7 it's possible to allow on a per module basis.

https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations

So leave it as it is for now.